src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -137,11 +148,42 @@ export const DataTable: FunctionComponent<Props> = ({
     });
   };
 
+  const headerCheckbox = (): React.ReactNode => {
+    function handleClick() {
+      handleHeaderCheckboxClick();
+    }
+    return (
+      <HeaderCheckbox
+        color="primary"
+        checked={isHeaderChecked}
+        onClick={handleClick}
+        indeterminate={isHeaderIndeterminant}
+      />
+    );
+  };
+
+  const rowCheckbox = (item: TableItem): React.ReactNode => {
+    let handleClick;
+    if (item !== undefined) {
+      const checked: boolean = checkedSamples.includes(item.publicId);
+      handleClick = function handleClick() {
+        handleRowCheckboxClick(
+          item.publicId.toString(),
+          Boolean(item.CZBFailedGenomeRecovery)
+        );
+      };
+      return (
+        <RowCheckbox color="primary" onClick={handleClick} checked={checked} />
+      );
+    }
+  };
+
   const render = (tableData: TableItem[]) => {
     function renderRow(props: ListChildComponentProps) {
       const item = tableData[props.index];
       return (
         <TableRow style={props.style} data-test-id="table-row">
+          {showCheckboxes && rowCheckbox(item)}
           {item === undefined ? null : sampleRow(item)}
```

Comment:
nit: positive case first for readability: `{ item ? sampleRow(item) : null }`

Reply:
yep that is much better! ty